tests/test_superruns.py

```diff
@@ -0,0 +1,247 @@
+import os
+import unittest
+import shutil
+import strax
+import numpy as np
+import tempfile
+from strax.testutils import Records, PeaksWoByRunDefault
+import datetime
+import pytz
+import json
+from bson import json_util
+import re
+
+
+class TestSuperRuns(unittest.TestCase):
+
+    def setUp(self, superrun_name='_superrun_test'):
+        self.offset_between_subruns = 10
+        self.superrun_name = superrun_name
+        self.tempdir = tempfile.mkdtemp()
+        self.context = strax.Context(storage=[strax.DataDirectory(self.tempdir,
+                                                                  provide_run_metadata=True,
+                                                                  readonly=False,
+                                                                  deep_scan=True)],
+                                     register=[Records, RecordsExtension],
+                                     use_per_run_defaults=False)
+        self.context.set_context_config({'write_superruns': True})
+        
+        logger = self.context.log
+        logger.addFilter(lambda s: not re.match(".*Could not estimate run start and end time.*",
+                                                s.getMessage()))
+
+        self._create_subruns()
+        self.context.define_run(self.superrun_name, data=self.subrun_ids)  # Define superrun
+
+    def test_run_meta_data(self):
+        """
+        Check if superrun has the correct run start/end and livetime.
+        """
```

Comment:
want to add an assert statement that the subruns are stored whereas the superrun is not prior to running the test. It kind of depends on the order of how the other tests are run so if you rename them or change the setup, you might run into issues.

Reply:
I do not think this is true. setUp and TearDown are called before every single test case so you are not running into these kind of things.

Reply:
Right, I see. You are right! I was confused for a second with setUpClass and tearDownClass for a second. Not that it hurts to check of course but okay.

This is the behavior I thought you envisaged, so my comment was assuming so 😂.

Hereby a purely academic example of how things would have worked with setUpClass and tearDownClass. This could decrease testing time if e.g. the setup itself takes a lot of time and can be reused.
```python
import tempfile
import os
import unittest
import shutil
import uuid


class TestTheTestingSuite(unittest.TestCase):
    """Test the per-run defaults options of a context"""
    @classmethod
    def setUpClass(cls) -> None:
        temp_folder = uuid.uuid4().hex
        cls.tempdir = os.path.join(tempfile.gettempdir(), temp_folder)
        assert not os.path.exists(cls.tempdir)

    @classmethod
    def tearDownClass(cls):
        shutil.rmtree(cls.tempdir)

    def test_one(self, make='1'):
        os.makedirs(os.path.join(self.tempdir, make))
        print(os.listdir(self.tempdir))

    def test_two(self):
        self.test_one('2')
```

```
tests/test_test.py::TestTheTestingSuite::test_one ['1']
PASSED
tests/test_test.py::TestTheTestingSuite::test_two ['1', '2']
PASSED
```

Reply:
Ah interesting I was not aware of this, but as you pointed out this requires some additional thinking :D Compared to some of our other tests I the superrun test are fairly fast. Hence, I will address this comment maybe later.

Reply:
For sure, it's only 8 seconds on actions. This example is purely academic on this point